java-frontend/src/main/java/org/sonar/java/se/xproc/MethodBehavior.java

```diff
@@ -153,6 +185,58 @@ public boolean isComplete() {
   public void completed() {
     this.complete = true;
     this.visited = true;
+    reduceYields();
+  }
+
+  private void reduceYields() {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 28 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3776)